executing/executing.py

```diff
@@ -509,9 +530,556 @@ def visit_ClassDef(self, node):
+        ):
+            return
+
+        if node_match(ast.Name, ctx=ast.Del) and inst_match(
+            ("DELETE_NAME", "DELETE_GLOBAL"), argval=mangled_name(node)
+        ):
+            return
+
+        # old verifier
+
+        typ = type(None)
+
+        if op_name.startswith(("BINARY_SUBSCR", "SLICE+")):
+            typ = ast.Subscript
+            ctx = ast.Load
+        elif op_name.startswith("BINARY_"):
+            typ = ast.BinOp
+            op_type = op_type_map[instruction.argrepr]
+            extra_filter = lambda e: isinstance(e.op, op_type)
+        elif op_name.startswith("UNARY_"):
+            typ = ast.UnaryOp
+            op_type = dict(
+                UNARY_POSITIVE=ast.UAdd,
+                UNARY_NEGATIVE=ast.USub,
+                UNARY_NOT=ast.Not,
+                UNARY_INVERT=ast.Invert,
+            )[op_name]
+            extra_filter = lambda e: isinstance(e.op, op_type)
+        elif op_name in ("LOAD_ATTR", "LOAD_METHOD", "LOOKUP_METHOD"):
+            typ = ast.Attribute
+            ctx = ast.Load
+            extra_filter = lambda e: attr_names_match(e.attr, instruction.argval)
+        elif op_name in (
+            "LOAD_NAME",
+            "LOAD_GLOBAL",
+            "LOAD_FAST",
+            "LOAD_DEREF",
+            "LOAD_CLASSDEREF",
+        ):
+            typ = ast.Name
+            ctx = ast.Load
+            if PY3 or instruction.argval:
+                extra_filter = lambda e: e.id == instruction.argval
+        elif op_name in ("COMPARE_OP", "IS_OP", "CONTAINS_OP"):
+            typ = ast.Compare
+            extra_filter = lambda e: len(e.ops) == 1
+        elif op_name.startswith(("STORE_SLICE", "STORE_SUBSCR")):
+            ctx = ast.Store
+            typ = ast.Subscript
+        elif op_name.startswith("STORE_ATTR"):
+            ctx = ast.Store
+            typ = ast.Attribute
+            extra_filter = lambda e: attr_names_match(e.attr, instruction.argval)
+
+        node_ctx = getattr(node, "ctx", None)
+
+        ctx_match = (
+            ctx is not type(None)
+            or not hasattr(node, "ctx")
+            or isinstance(node_ctx, ctx)
+        )
+
+        # check for old verifier
+        if isinstance(node, typ) and ctx_match and extra_filter(node):
+            return
+
+        # generate error
+
+        title = "ast.%s is not created from %s" % (
+            type(node).__name__,
+            instruction.opname,
+        )
+
+        raise VerifierFailure(title, node, instruction)
+
+    def instruction(self, index):
+        return self.bc_list[index // 2]
+
+    def opname(self, index):
+        return self.instruction(index).opname
+
+    def find_node(
+        self,
+        index,
+        match_positions=("lineno", "end_lineno", "col_offset", "end_col_offset"),
+        typ=None,
+    ):
+        if typ == None:
+            # this is not a default argument because ast.MatchAs does not exist in older python versions
```

Comment:
I assume this is meant to say `ast.pattern`. But patterns and nodes in patterns are rejected anyway, right?

Reply:
Yes. We have to identify the node to generate a `KnownIssue`, which then gets ignored.